symbolic-symcache/src/writer.rs

```diff
@@ -381,8 +381,17 @@ where
             return Ok(*index);
         }
 
+        // Instead of failing hard when exceeding the maximum allowed number of files (u16::MAX),
+        // we rather insert a sentinel "<unknown>" file and use that for all files that exceed the max.
+        const MAX_FILE_INDEX: u16 = std::u16::MAX - 1;
         if self.files.len() >= std::u16::MAX as usize {
-            return Err(SymCacheErrorKind::TooManyValues(ValueKind::File).into());
+            return Ok(MAX_FILE_INDEX);
```

Comment:
shouldn't this return `std::u16::MAX`?  I think now you're returning the index of the last real entry, instead of the one with `<unknown>`
(btw, should you do `u16::MAX` these days?  i thought clippy would be unhappy but maybe it doesn't matter)

Reply:
lol, I was stupid, yes. being at `MAX` means we add the sentinel and return it, being over MAX means we always return the sentinel index (which is MAX)